library/manageiq.py

```diff
@@ -64,14 +61,110 @@
+
+    def __init__(self, module, url, user, password):
+        self.module        = module
+        self.api_url       = url + '/api'
+        self.user          = user
+        self.password      = password
+        self.client        = MiqApi(self.api_url, (self.user, self.password))
+        self.changed       = False
+        self.msg           = ''
+        self.providers_url = self.api_url + '/providers'
+
+    def update_required(self, provider_id, hostname, port):
+        """ Checks whether an update is required for the provider
+
+        Returns:
+            False if the hostname and port passed equals the provider's,
+            True otherwise
+        """
+        try:
+            result = self.client.get(self.providers_url + '/%d/?attributes=authentications,endpoints' % provider_id)
+        except Exception as e:
+            self.msg = "Failed to get provider data. Error: %s" % e
+            self.module.fail_json(msg=self.msg)
         endpoint = result['endpoints'][0]
         return False if (endpoint['hostname'] == hostname and endpoint['port'] == int(port)) else True
-    except Exception as e:
-        msg = "Failed to get provider data. Error: %s" % e
-        module.fail_json(msg=msg)
+
+    def update_provider(self, provider_id, provider_name, endpoints):
+        """ Updates the existing provider with new parameters
+        """
+        try:
+            result = self.client.post(self.providers_url + '/%d' % provider_id,
+                                      action='edit',
+                                      connection_configurations=endpoints)
+            self.msg = "Successfuly updated %s provider" % provider_name
+            self.changed = True
+        except Exception as e:
+            self.msg = "Failed to update provider. Error: %s" % e
+            self.module.fail_json(msg=self.msg)
+
+    def add_new_provider(self, provider_name, endpoints):
+        """ Adds a provider to manageiq
+
+        Returns:
+            the added provider id
+        """
+        try:
+            result = self.client.post(self.providers_url, type=ManageIQ.openshift_provider_type,
+                                      name=provider_name,
+                                      connection_configurations=endpoints)
+            provider_id = result['results'][0]['id']
+            self.msg = "Successfuly added %s provider" % provider_name
+            self.changed = True
+        except Exception as e:
+            self.msg = "Failed to add provider. Error: %s" % e
+            self.module.fail_json(msg=self.msg)
+        return provider_id
+
+    def find_provider_by_name(self, provider_name):
+        """ Searches the provider name in manageiq existing providers
+
+        Returns:
+            the provider id if it exists in manageiq, None otherwise
+        """
+        providers = self.client.collections.providers
+        for provider in providers:
```

Comment:
Neat!  But even though I know how this works, I find the existing for: if: return code more readable.